### ISSUE TITLE
[BUG] Bot Keeps Transfering High IV Pokemon

### ISSUE THREAD

**Issue by Pakobbix** [NecronomiconCoding/NecroBot#301]
The Bot ignores now the min Keep IV value and the PrioritizeIVoverCP and transfer all duplicate Pokemons with lower CP.

[23:47:29] Jynx with 334 (98,07 % perfect) CP (Best: 772 | (90,99 % perfect))

[00:13:43] Krabby with 377 (92,91 % perfect) CP (Best: 510 | (61,61 % perfect))

Settings: Keep 90% IV
1400 CP
PrioritizeIVoverCP = True

**Comment by navidemad** [comment]
![hitmon](https://cloud.githubusercontent.com/assets/6686865/17080575/f99071f0-50e9-11e6-9a65-859176012d75.png)

I have also a similar problem,

**Comment by ridoknight** [comment]
Yeah i same here. I am using latest version and it sent my %97 magicarp :((

**Comment by qwikwit** [comment]
Same here

**Comment by Shishiaha** [comment]
Same here. Had to stop it right away.

Thanks for resolving this issue.

**Comment by PocketMobsters** [comment]
PrioritizeIVOverCP is a boolean value, which means the only values it can hold are "True" and "False" if you set it to "90%" like your post is saying, it might do horrible things that we never tested for. Keep within the bounds of the values if possible.

**Comment by ridoknight** [comment]
I chose "True" but bot sent high IV poke to prof.

**Comment by Shishiaha** [comment]
Even with this setting off, it do not keep pokémon >MinIVs.

Going to look closer in code what changed.

**Comment by Pakobbix** [comment]
@PocketMobsters Yeah sry... was a brainfart with 90% .. it's set to True

**Comment by PocketMobsters** [comment]
The only thing my code did was change ordering by IV instead of CP if it's toggle to true, it doesn't change anything else so if you're having trouble with your min IVs it's not due to that code.

**Comment by qwikwit** [comment]
taking a closer look into the code

**Comment by Shishiaha** [comment]
I need to see old code to compare but can't find a reference to MinIV in _inventory.

I deleted old source from my computer -.-

Can someone paste old source for _inventory.GetDuplicatePokemonToTransfer

**Comment by bioid** [comment]
You can look at the history of the file on github: https://github.com/NecronomiconCoding/NecroBot/commits/master/PokemonGo.RocketAPI.Logic/Inventory.cs

**Comment by Citrusbomb** [comment]
![untitled](https://cloud.githubusercontent.com/assets/20583237/17080786/610e6a0a-513d-11e6-8e1c-886385acd104.png)

Its transferring high IV and it seems like its also comparing the pokemon its transferring as its technically in the inventory?

**Comment by qwikwit** [comment]
I recommend to delete code. Its easier to keep all mon that fit "KeepMinIVPercentage"

**Comment by ridoknight** [comment]
@qwikwit Yeah i want to delete it to.

**Comment by Shishiaha** [comment]
hum 


https://github.com/NecronomiconCoding/NecroBot/commit/193e4cafa71a6872f6dffc6fa0e3b89718f82614

**Comment by bebar69** [comment]
yeah IV Eevvee 100% drop ;(

**Comment by Shishiaha** [comment]
```
foreach (var duplicatePokemon in duplicatePokemons)
             {
**# if (PokemonInfo.CalculatePokemonPerfection(duplicatePokemon) >= _clientSettings.KeepMinIVPercentage || duplicatePokemon.Cp > _clientSettings.KeepMinCP)
continue;**

                 var transfer = await _client.TransferPokemon(duplicatePokemon.Id);
                 _stats.IncreasePokemonsTransfered();
```

Update this in Logic.cs

**Comment by qwikwit** [comment]
![image](https://cloud.githubusercontent.com/assets/20604817/17080937/188c3f90-5111-11e6-8852-846be06b8693.png)
it does a double try.

**Comment by Pakobbix** [comment]
PocketMobster already made a PR and Necro merged it 20mins ago.. I'm already testing it and report back (as soon as i loose another good IV pokemon or not ^^)

**Comment by Shishiaha** [comment]
Ok, for now i stay with 1.0.4.0 and this workaround.

Wait ur loss ;)

**Comment by Pakobbix** [comment]
[02:19:15] (TRANSFERED) Rattata with 155 (78,15 % perfect) CP (Best: 371 | (76,32 % perfect))

reduced the IV perfection to 70% so it should keep the 78,15%.. but it still exists

**Comment by Citrusbomb** [comment]
Still dropping high IVs

**Comment by Citrusbomb** [comment]
![untitled](https://cloud.githubusercontent.com/assets/20583237/17080974/4b42c362-5145-11e6-8c8f-20a87cbcd0a2.png)

**Comment by Shishiaha** [comment]
Just use my comment for now then...

> foreach (var duplicatePokemon in duplicatePokemons)
>              {
>**if (PokemonInfo.CalculatePokemonPerfection(duplicatePokemon) >= _clientSettings.KeepMinIVPercentage || duplicatePokemon.Cp > _clientSettings.KeepMinCP)
> continue;**
> 
>var transfer = await _client.TransferPokemon(duplicatePokemon.Id);
>_stats.IncreasePokemonsTransfered();

Update this in Logic.cs

It's not the best but who cares.

**Comment by Pakobbix** [comment]
@Citrusbomb  If you don't set your KeepIV setting to 20% the Oddish is normal (gets transfered cause the other one got higher CP) 

So only Kabuto is your loss

@Shishiaha gonna try it.. gimme a sec + hrs for testing xD

**Comment by Shishiaha** [comment]
It works. Doing it.

I have some 100% i don't even wan"t to loose now.

**Comment by Citrusbomb** [comment]
@Pakobbix But if IV>CP then the 687 should have been transfered?

**Comment by Citrusbomb** [comment]
@Shishiaha Testing

**Comment by Pakobbix** [comment]
@Citrusbomb I think the Priority first kicks in at KeepIV setting.. dunno.. we'll see if it works (or not and necro removes the "feature" again)

**Comment by qwikwit** [comment]
prioritizeIVoverCP vs prioritizeIVOverCP

**Comment by Pakobbix** [comment]
@Shishiaha IT WORKS!!! yay :-D It now keeps both (Highest CP and highest IV) but i don't care.. atleast my High IV are safe for now :-)

Thank you very much.

**Comment by TrYkeRm4N** [comment]
so Pakobbix, you fix it doing the thing that Shishiaha  said?

**Comment by Pakobbix** [comment]
Yep.. it's a workaround.. for now.. the guys on Discord looking for an permanent solution (Something messed up in Inventory.cs

**Comment by Citrusbomb** [comment]
Works for me, Thanks Shish

**Comment by TrYkeRm4N** [comment]
Damn, I'm on my parents house, got no visual studio here, can a mercy soul upload a compiled version (windows) to dropbox or similar? with the Shishiaha fix?

@Citrusbomb 
@Pakobbix

**Comment by Pakobbix** [comment]
resolved #301

**Comment by ReddoKiddo** [comment]
Seems to be fixed and @TrYkeRm4N no compiled version is allowed in our repo/discord
